CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -1,22 +1,97 @@
 ﻿using System;
+using AutoMapper;
 using System.Collections.Generic;
 using System.Security.Claims;
 using System.Security.Principal;
 using System.Threading.Tasks;
 using CharlieBackend.Business.Services;
 using CharlieBackend.Core.DTO.Dashboard;
 using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Mapping;
 using CharlieBackend.Core.Models.ResultModel;
 using CharlieBackend.Data.Repositories.Impl.Interfaces;
 using Moq;
 using Xunit;
+using FluentAssertions;
 
 namespace CharlieBackend.Api.UnitTest
 {
     public class DashboardServiceTest : TestBase
     {
+        private readonly Mock<IDashboardRepository> _dashboardRepositoryMock;
+        private readonly IMapper _mapper;
+        private static long studentIdWithGroup = 7;
+        private static long studentIdWithoutGroup = 20;
+        private static long courseId = 1;
+        private static long courseIdWithoutGroup = 2;
+
+        public DashboardServiceTest()
+        {
+            _dashboardRepositoryMock = new Mock<IDashboardRepository>();
+            _mapper = GetMapper(new ModelMappingProfile());
+        }
+
+        private void InitializeForDashboardRepository()
+        {
+            _unitOfWorkMock.Setup(x => x.DashboardRepository).Returns(_dashboardRepositoryMock.Object);
+            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
+        }
+
+        [Fact]
+        public async Task GetStudentsClassbook_ValidDataPassed_ShouldReturnCurrentStudentsMarks()
```

Comment:
Please rename this test to GetStudentsClassbook_ValidDataPassed_ShouldReturnExpectedData and move verification of Presences to this tests as we have to test possible conditions by one method but not returned property.

Reply:
Changed in this commit https://github.com/ita-social-projects/WhatBackend/pull/409/commits/1aeb8fbf56772c98a76b07f66be14b5bf1610399